pkg/codegen/go/gen.go

```diff
@@ -1323,6 +1332,86 @@ func (pkg *pkgContext) genConfig(w io.Writer, variables []*schema.Property) erro
 	return nil
 }
 
+// genResourceModule generates a ResourceModule definition and the code to register an instance thereof with the
+// Pulumi runtime. The generated ResourceModule supports the deserialization of resource references into fully-
+// hydrated Resource instances. If this is the root module, this function also generates a ResourcePackage
+// definition and its registration to support rehydrating providers.
+func (pkg *pkgContext) genResourceModule(w io.Writer) {
+	contract.Assert(len(pkg.resources) != 0)
+
+	imports := codegen.NewStringSet(
+		"github.com/blang/semver",
+		"github.com/pulumi/pulumi/sdk/v2/go/pulumi",
+	)
+
+	// Check for provider-only modules.
+	if providerOnly := len(pkg.resources) == 1 && pkg.resources[0].IsProvider; providerOnly {
+		pkg.genHeader(w, []string{"fmt"}, imports)
+
+		fmt.Fprintf(w, "type module struct {\n")
+		fmt.Fprintf(w, "\tversion semver.Version\n")
+		fmt.Fprintf(w, "}\n\n")
+
+		fmt.Fprintf(w, "func (m *module) Version() semver.Version {\n")
+		fmt.Fprintf(w, "\treturn m.version\n")
+		fmt.Fprintf(w, "}\n\n")
+
+		fmt.Fprintf(w, "func (m *module) ConstructProvider(ctx *pulumi.Context, name, typ, urn string) (pulumi.ProviderResource, error) {\n")
+		fmt.Fprintf(w, "\tif typ != \"pulumi:providers:%s\" {\n", pkg.pkg.Name)
+		fmt.Fprintf(w, "\t\treturn nil, fmt.Errorf(\"unknown provider type: %%s\", typ)\n")
+		fmt.Fprintf(w, "\t}\n\n")
+		fmt.Fprintf(w, "\treturn NewProvider(ctx, name, nil, pulumi.URN_(urn))\n")
+		fmt.Fprintf(w, "}\n\n")
+
+		fmt.Fprintf(w, "func init() {\n")
+		fmt.Fprintf(w, "\tversion, err := PkgVersion()\n")
+		fmt.Fprintf(w, "\tif err != nil {\n")
+		fmt.Fprintf(w, "\t\tpanic(fmt.Errorf(\"failed to determine package version: %%v\", err))\n")
+		fmt.Fprintf(w, "\t}\n")
+		fmt.Fprintf(w, "\tpulumi.RegisterResourcePackage(\n")
+		fmt.Fprintf(w, "\t\t%q,\n", pkg.pkg.Name)
+		fmt.Fprintf(w, "\t\t&module{version},\n")
+		fmt.Fprintf(w, "\t)\n")
+		fmt.Fprintf(w, "}\n")
+	} else {
+		imports.Add("github.com/pulumi/pulumi-kubernetes/sdk/v2/go/kubernetes")
```

Comment:
Is this import of the k8s SDK intentional?

Reply:
Nope - it's fixed in the updates I just pushed